### PR TITLE
Closes #201 | Implemente dataloader for VISTEC_TP_TH_21

### LINKED ISSUE #201: Create dataset loader for VISTEC-TP-TH-21
Dataloader name: `vistec_tp_th_21/vistec_tp_th_21.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?vistec_tp_th_21

|     Dataset| vistec_tp_th_21 |
|-------------|---|
| Description | The largest social media domain datasets for Thai text processing (word segmentation, misspell correction and detection, and named-entity boundary) called "VISTEC-TP-TH-2021" or VISTEC-2021. VISTEC corpus contains 49,997 sentences with 3.39M words where the collection was manually annotated by linguists on four tasks, namely word segmentation, misspelling detection and correction, and named entity recognition. |
| Subsets | - |
| Languages | tha |
| Tasks | Named Entiy Recognition |
| License | Creative Commons Attribution Share Alike 3.0 (cc-by-sa-3.0) |
| Homepage | https://github.com/mrpeerat/OSKut/tree/main/VISTEC-TP-TH-2021 |
| HF URL | - |
| Paper URL | https://aclanthology.org/2021.findings-acl.86.pdf |

### PR DESCRIPTION
**Pull request by Alex-HaochenLi**
Closes #201 .

Note that this dataset has a specific label for NER. It only contains True/False labels for NER.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/vistec_tp_th_21/vistec_tp_th_21.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class VISTEC21(datasets.GeneratorBasedBuilder):
```
Comment (jamesjaya) [inline seacrowd/sea_datasets/vistec_tp_th_21/vistec_tp_th_21.py:70]: Please add Dataset suffix to the class name

```suggestion
class VISTEC21Dataset(datasets.GeneratorBasedBuilder):
```

File: seacrowd/sea_datasets/vistec_tp_th_21/vistec_tp_th_21.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+_DESCRIPTION = """\
+The largest social media domain datasets for Thai text processing (word segmentation, 
+misspell correction and detection, and named-entity boundary) called "VISTEC-TP-TH-2021" or VISTEC-2021. 
+VISTEC corpus contains 49,997 sentences with 3.39M words where the collection was manually annotated by 
+linguists on four tasks, namely word segmentation, misspelling detection and correction, 
+and named entity recognition.
+"""
+
+_HOMEPAGE = "	https://github.com/mrpeerat/OSKut/tree/main/VISTEC-TP-TH-2021"
```
Comment (jamesjaya) [inline seacrowd/sea_datasets/vistec_tp_th_21/vistec_tp_th_21.py:48]: remove \t prefix
```suggestion
_HOMEPAGE = "https://github.com/mrpeerat/OSKut/tree/main/VISTEC-TP-TH-2021"
```

File: seacrowd/sea_datasets/vistec_tp_th_21/vistec_tp_th_21.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+_CITATION = """\
+@inproceedings{limkonchotiwat2021handling,
+  title={Handling Cross-and Out-of-Domain Samples in Thai Word Segmentation},
+  author={Limkonchotiwat, Peerat and Phatthiyaphaibun, Wannaphong and Sarwar, Raheem and Chuangsuwanich, Ekapol and Nutanong, Sarana},
+  booktitle={Findings of the Association for Computational Linguistics: ACL-IJCNLP 2021},
+  pages={1003--1016},
+  year={2021}
+}
+"""
```
[the remark this answers is not in the document]
Comment (Alex-HaochenLi) [reply seacrowd/sea_datasets/vistec_tp_th_21/vistec_tp_th_21.py]: Have revised all of your reviews and committed. Please have a check :)

**Review by SamuelCahyawijaya — CHANGES_REQUESTED** [review]
Hi @Alex-HaochenLi , thank you for contributing. The dataset works as expected. LGTM!

**Review by SamuelCahyawijaya — APPROVED** [review]

**Pull request merged**